Backend/src/main/java/ktsnvt/tim1/services/EventService.java

```diff
@@ -148,12 +177,13 @@ private void enableEventGroups(Event e, Location l, LocationSeatGroupDTO seatGro
                 EventSeatGroup esg = new EventSeatGroup();
                 esg.setPrice(esgDTO.getPrice());
                 esg.setSeatGroup(seatGroup);
-                if (seatGroup.getParterre()) {
-                    esg.setFreeSeats(esgDTO.getFreeSeats());
-                }
-                else {
-                    esg.setFreeSeats(seatGroup.getColsNum() * seatGroup.getRowsNum());
-                }
+                e.getEventDays().forEach(eventDay -> {
+                    ReservableSeatGroup rsg = new ReservableSeatGroup();
+                    rsg.setEventDay(eventDay);
+                    rsg.setEventSeatGroup(esg);
+                    rsg.setFreeSeats(esg.getSeatGroup().getTotalSeats());
```

Comment:
There should a parametrised ReservableSeatGroup constructor that takes EventDay, EventSeatGroup, and totalSeatNumber. Furthermore, in that constructor Seat objects are to be made, if corresponding SeatGroup object is not a parterre.